time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeries.java

```diff
@@ -288,11 +340,40 @@ private Duration checkRegularSpacing() {
                     StringDataChunk chunk = new UncompressedStringDataChunk(0, stringValues.toArray(new String[0])).tryToCompress();
                     timeSeriesList.add(new StringTimeSeries(metadata, chunk));
                 } else {
-                    throw assertDataType(dataTypes[i - 2]);
+                    throw assertDataType(dataTypes[i - fixedColumns]);
                 }
             }
             return timeSeriesList;
         }
+
+        private TimeSeriesIndex getTimeSeriesIndex() {
+            Long spacing = checkRegularSpacing();
+            if (spacing != Long.MIN_VALUE) {
+                return new RegularTimeSeriesIndex(times.get(0), times.get(times.size() - 1), spacing);
+            } else {
+                return new IrregularTimeSeriesIndex(times.stream().mapToLong(l -> l).toArray());
+            }
+        }
+
+        private long checkRegularSpacing() {
+            if (times.size() < 2) {
+                throw new TimeSeriesException("At least 2 rows are expected");
+            }
+
+            long spacing = Long.MIN_VALUE;
+            for (int i = 1; i < times.size(); i++) {
+                long duration = times.get(i) - times.get(i - 1);
+                if (spacing == Long.MIN_VALUE) {
+                    spacing = duration;
+                } else {
+                    if (1e-4 < Math.abs(duration - spacing)) {
+                        return Long.MIN_VALUE;
+                    }
```

Comment:
What did you have in mind by replacing the equality by this epsilon? Maybe you should also introduce a constant instead of a magic number. What is the unit of this value?

Reply:
Replaced with an equality